src/commands/explorer.ts

```diff
@@ -199,294 +266,386 @@ export class ExplorerCommand extends BaseCommand {
   }
 
   private async prepareValuesArg(config: ExplorerDeployConfigClass): Promise<string> {
-    let valuesArgument = '';
+    let valuesArgument: string = '';
     if (config.valuesFile) {
       valuesArgument += prepareValuesFiles(config.valuesFile);
     }
     return valuesArgument;
   }
 
+  private installCertManagerTask(): SoloListrTask<AnyListrContext> {
+    return {
+      title: 'Install cert manager',
+      skip: ({config}: ExplorerDeployContext | ExplorerUpgradeContext): boolean => !config.enableExplorerTls,
+      task: async ({config}: ExplorerDeployContext | ExplorerUpgradeContext): Promise<void> => {
+        config.soloChartVersion = Version.getValidSemanticVersion(config.soloChartVersion, false, 'Solo chart version');
+        const {soloChartVersion} = config;
+
+        const soloCertManagerValuesArgument: string = await this.prepareCertManagerChartValuesArg(config);
+        // check if CRDs of cert-manager are already installed
+        let needInstall: boolean = false;
+        for (const crd of constants.CERT_MANAGER_CRDS) {
+          const crdExists: boolean = await this.k8Factory.getK8(config.clusterContext).crds().ifExists(crd);
+          if (!crdExists) {
+            needInstall = true;
+            break;
+          }
+        }
+
+        if (needInstall) {
+          // if cert-manager isn't already installed we want to install it separate from the certificate issuers
+          // as they will fail to be created due to the order of the installation being dependent on the cert-manager
+          // being installed first
+          await this.chartManager.install(
+            NamespaceName.of(constants.CERT_MANAGER_NAME_SPACE),
+            constants.SOLO_CERT_MANAGER_CHART,
+            constants.SOLO_CERT_MANAGER_CHART,
+            config.chartDirectory ? config.chartDirectory : constants.SOLO_TESTING_CHART_URL,
+            soloChartVersion,
+            '  --set cert-manager.installCRDs=true',
+            config.clusterContext,
+          );
+          showVersionBanner(this.logger, constants.SOLO_CERT_MANAGER_CHART, soloChartVersion);
+        }
+
+        // wait cert-manager to be ready to proceed, otherwise may get error of "failed calling webhook"
+        await this.k8Factory
+          .getK8(config.clusterContext)
+          .pods()
+          .waitForReadyStatus(
+            constants.DEFAULT_CERT_MANAGER_NAMESPACE,
+            ['app.kubernetes.io/component=webhook', `app.kubernetes.io/instance=${constants.SOLO_CERT_MANAGER_CHART}`],
+            constants.PODS_READY_MAX_ATTEMPTS,
+            constants.PODS_READY_DELAY,
+          );
+
+        // sleep for a few seconds to allow cert-manager to be ready
+        sleep(Duration.ofSeconds(10));
```

Comment:
Should this only happen if we are installing cert-manager? Based on the code it appears we do this check every time and assume that an externally managed cert-manager is using the default namespace. 

Wouldn't it be better to assume that if the CRDs already exist then cert-manager is already installed and available?